src/lib.rs

```diff
@@ -1,4 +1,6 @@
 #![doc = include_str!("../README.md")]
+//! ## Crate feature flags
+#![doc = document_features::document_features!()]
```

Comment:
I'd also like to enable `doc_auto_cfg` for the fancy `cargo doc` callouts, but that only works on nightly.  Seems like the workaround is to use a build.rs file to detect the compiler version and enable it on nightly only.  Seems a bit messy, but I'm happy to add that.